GitPrompt.ps1

```diff
@@ -145,13 +145,31 @@ function Write-GitStatus($status) {
     }
 }
 
-if(!(Test-Path Variable:Global:VcsPromptStatuses)) {
-    $Global:VcsPromptStatuses = @()
+$VcsPromptStatusesVariableName = "VcsPromptStatuses"
+
+if (-not (Test-Path Variable:Global:$VcsPromptStatusesVariableName)) {
+    New-Variable -Name $VcsPromptStatusesVariableName -Value @() -Scope Global
+}
+
+function Global:Write-VcsStatus {
+    Get-Variable -Name $VcsPromptStatusesVariableName -Scope Global -ValueOnly | foreach { 
+        try {
+            $_.Invoke()
+        }
+        catch {
+            Write-Error "A Write-VcsStatus scriptblock failed with the error: $($_.Exception.InnerException)"

```

Comment:
Since this is really meant to be used in a prompt, I wonder if it would be less intrusive to replace `Write-Error` with an inline message that looks like an error (i.e. `Write-Host '[error]' -NoNewLine -ForegroundColor Red -BackgroundColor Black`)?